### PR TITLE
Base64 data issue/v4

### PR DESCRIPTION
**Pull request by inashivb**
New PR with better commits coming up. This is to run the QA.

### CODE REVIEW

File: src/util-base64.c [hunk 1]
```diff
@@ -156,9 +156,25 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
...
+                            : 2; // Decoded bytes for 1 or 2 base64 encoded bytes is 1
+        uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
+        if (dest_size < *decoded_bytes + numDecoded_blk) {
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            SCLogNotice("consumed_bytes: %d, decoded_bytes: %d", *consumed_bytes, *decoded_bytes);
+            return ecode;
+        }
+        if (dest_size % 3 !=
```
Comment (victorjulien) [inline src/util-base64.c:171]: what is the logic of having the destination size be a multiple of 3?

Comment (inashivb) [reply src/util-base64.c:171]: It should not be less than 3. This is wrong.

Comment (inashivb) [reply src/util-base64.c:171]: ok. I'm thinking about how to handle this in the callers.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 145160 | 113246 | 78.01% | 

Pipeline 12933

**Comment by victorjulien** [comment]
Merged https://github.com/OISF/suricata/pull/8651/commits/f873a96f1be3932ef3eed4f385bd2127ea866462, thanks.

**Comment by inashivb** [comment]
Replaced w #8652

**Pull request closed without merging**